app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb

```diff
@@ -20,7 +20,11 @@ def flavors
   end
 
   def images
-    connection.images.all
+    @images ||= connection.images.all.to_a
+  end
+
+  def images_by_id
+    @images_by_id ||= images.index_by { |img| img[:id] }
```

Comment:
Is it a best practice to build these accessibility maps in the collector rather than the parser?  Makes sense.

Reply:
Yeah this is commonly where we do this sort of thing, it isn't strictly required but when we start doing targeted refresh doing this in the Collector class gives us some freedom on building these hashes without changing the Parser at all.